File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/agent/AgentRemoveIdentityTest.java
package org.hyperledger.indy.sdk.agent;

import org.hyperledger.indy.sdk.agent.Agent.Listener;
import org.hyperledger.indy.sdk.signus.Signus;
import org.hyperledger.indy.sdk.signus.SignusResults;
import org.junit.Test;


public class AgentRemoveIdentityTest extends AgentIntegrationTest {

	@Test
	public void testAgentRemoveIdentityWorks() throws Exception {
		String endpoint = "127.0.0.1:9608";

		SignusResults.CreateAndStoreMyDidResult myDid = Signus.createAndStoreMyDid(wallet, "{}").get();

		Listener activeListener = Agent.agentListen(endpoint, incomingConnectionObserver).get();

		activeListener.agentAddIdentity(pool, wallet, myDid.getDid()).get();

		activeListener.agentRemoveIdentity(wallet, myDid.getDid()).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/AgentTests/AgentRemoveIdentityTest.cs
using Hyperledger.Indy.AgentApi;
using Hyperledger.Indy.SignusApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AgentTests
{
    [TestClass]
    public class AgentRemoveIdentityTest : AgentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestAgentRemoveIdentityWorks()
        {
            var endpoint = "127.0.0.1:9608";

            var myDidResult = await Signus.CreateAndStoreMyDidAsync(wallet, "{}");

            var activeListener = await AgentListener.ListenAsync(endpoint);
            await activeListener.AddIdentityAsync(pool, wallet, myDidResult.Did);

            await activeListener.RemoveIdentityAsync(wallet, myDidResult.Did);
        }
    }
}
